src/main/java/org/folio/rest/impl/PatronServicesResourceImpl.java

```diff
@@ -82,10 +85,109 @@
+          return responseJson.getJsonArray("usergroups").getJsonObject(0).getString("id");
+        } else {
+          throw new IllegalArgumentException(new HttpException(500, "Remote patron group not found"));
+        }
+      });
+  }
+
+  private CompletableFuture<JsonObject> getAddressTypes(UserRepository userRepository, Map<String, String> okapiHeaders) {
+    return userRepository.getAddressByType(okapiHeaders)
+      .thenApply(responseJson -> {
+        JsonObject addressTypes = new JsonObject();
+        responseJson.getJsonArray("addressTypes").forEach(item -> {
+          JsonObject addressType = (JsonObject) item;
+          if ("Home".equalsIgnoreCase(addressType.getString("addressType"))) {
+            addressTypes.put("home", addressType.getString("id"));
+          } else if ("Work".equalsIgnoreCase(addressType.getString("addressType"))) {
+            addressTypes.put("work", addressType.getString("id"));
+          }
+        });
+        return addressTypes;
+      });
+  }
+
+  private CompletableFuture<Integer> getUserByEmail(String email, Map<String, String> okapiHeaders, UserRepository userRepository) {
+    return userRepository.getUserByEmail(email, okapiHeaders)
+      .thenApply(responseJson -> responseJson.getInteger(TOTAL_RECORDS));
+  }
+
+  private CompletableFuture<Response> handleUserRecords(int totalRecords, ExternalPatron entity, Map<String, String> okapiHeaders, UserRepository userRepository, String remotePatronGroupId, String homeAddressTypeId, String workAddressTypeId) {
+    if (totalRecords > 1) {
+      return CompletableFuture.completedFuture(
+        PostPatronAccountResponse.respond400WithTextPlain("Multiple users found with the same email")
+      );
+    } else if (totalRecords == 1) {
+      return userRepository.getUserByEmail(entity.getContactInfo().getEmail(), okapiHeaders)
+        .thenApply(responseJson -> responseJson.getJsonArray("users").getJsonObject(0))
+        .thenCompose(userJson -> processSingleUser(userJson, remotePatronGroupId));
+    } else {
+      return createUser(entity, okapiHeaders, userRepository, remotePatronGroupId, homeAddressTypeId, workAddressTypeId);
+    }
+  }
+
+  private CompletableFuture<Response> processSingleUser(JsonObject userJson, String remotePatronGroupId) {
+    boolean isActive = userJson.getBoolean("active", false);
+    String patronGroup = userJson.getString("patronGroup", "");
+
+    if (!isActive) {
+      return CompletableFuture.completedFuture(
+        PostPatronAccountResponse.respond403WithTextPlain("User account is not active")
+      );
+    } else if (remotePatronGroupId.equals(patronGroup)) {
+      return CompletableFuture.completedFuture(
+        PostPatronAccountResponse.respond403WithTextPlain("User already exists")
+      );
+    } else {
+      return CompletableFuture.completedFuture(
+        PostPatronAccountResponse.respond403WithTextPlain("User does not belong to the required patron group")
+      );
+    }
```

Comment:
Pls check the error response code and error message.

Reply:
fixed